src/contour/Config.h

```diff
@@ -144,263 +189,3020 @@ struct SimpleColorConfig
+        loadFromEntry("word_delimiters", c.wordDelimiters);
+
+        loadFromEntry("read_buffer_size", c.ptyReadBufferSize);
+        loadFromEntry("pty_buffer_size", c.ptyBufferObjectSize);
+        loadFromEntry("images.sixel_register_count", c.maxImageColorRegisters);
+
+        loadFromEntry("live_config", c.live);
+        loadFromEntry("spawn_new_process", c.spawnNewProcess);
+        loadFromEntry("images.sixe_scrolling", c.sixelScrolling);
+        loadFromEntry("reflow_on_resize", c.reflowOnResize);
+        loadFromEntry("renderer.tile_direct_mapping", c.textureAtlasDirectMapping);
+
+        loadFromEntry("renderer.tile_hastable_slots", c.textureAtlasHashtableSlots);
+        loadFromEntry("renderer.tile_cache_count", c.textureAtlasTileCount);
+
+        loadFromEntry("bypass_mouse_protocol_modifier", c.bypassMouseProtocolModifier);
+
+        loadFromEntry("on_mouse_select", c.onMouseSelection);
+        loadFromEntry("mouse_block_selection_modifier", c.mouseBlockSelectionModifier);
+
+        loadFromEntry("input_mapping", c.inputMappings);
+
+        loadFromEntry("images", c.maxImageSize);
+
+        loadFromEntry("", c.experimentalFeatures);
+
+        loadFromEntry("profiles", c.profiles);
+
+        loadFromEntry("color_schemes", c.colorschemes);
+    };
+};
+
+struct YAMLConfigWriter
+{
+
+    std::string addOffset(string doc, size_t off)
+    {
+        auto offset = std::string(off, ' ');
+        return std::regex_replace(doc, std::regex(".+\n"), offset + "$&");
+    }
+
+    string createString(Config& c)
+    {
+        string doc {};
+        int level { 0 };
+        auto process = [&](auto v) {
+            doc.append(format(addOffset(v.documentation(), level * 4), v.get()));
+        };
+
+        auto processWithDoc = [&](auto docString, auto... val) {
+            doc.append(fmt::format(fmt::runtime(addOffset(docString, level * 4)), val...));
+        };
+
+        doc.append("START OF DOCUMENTATION \n");
+        process(c.platformPlugin);
+
+        // inside rendered:
+        level++;
+        doc.append("rendered: \n");
+        process(c.renderingBackend);
+        process(c.textureAtlasDirectMapping);
+        process(c.textureAtlasHashtableSlots);
+        process(c.textureAtlasTileCount);
+        level--;
+
+        process(c.wordDelimiters);
+        process(c.ptyReadBufferSize);
+        process(c.ptyBufferObjectSize);
+        process(c.defaultProfileName);
+        process(c.spawnNewProcess);
+        process(c.reflowOnResize);
+        process(c.bypassMouseProtocolModifier);
+        process(c.mouseBlockSelectionModifier);
+        process(c.onMouseSelection);
+        process(c.live);
+        process(c.experimentalFeatures);
+
+        // inside images:
+        doc.append("\nimages: \n");
+        level++;
+        process(c.sixelScrolling);
+        process(c.maxImageColorRegisters);
+        process(c.maxImageSize);
+        level--;
```

Comment:
maybe we can use some RAII helper to avoid `x++; ...; x--;` that would be a bit more error prone.